lib/llm/provider.rb

```diff
@@ -14,10 +14,10 @@ class Provider
     #  The host address of the LLM provider
     # @param [Integer] port
     #  The port number
-    def initialize(secret, host, port = 443)
+    def initialize(secret, host, port = 443, ssl: true)
```

Comment:
Can we make `host`, and `port` keyword arguments too ? Everything given to us via `options` should be a keyword argument.